File: submitted_code/100294.lian-biao-zhong-dao-shu-di-kge-jie-dian-lcof.cpp
//
// @lc app=leetcode.cn id=100294 lang=cpp
//
// [100294] lian-biao-zhong-dao-shu-di-kge-jie-dian-lcof
//
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* getKthFromEnd(ListNode* head, int k) {
        int i=0;
        ListNode *p = head;
        while(p && i<k){
            p=p->next;
            i++;
        }
        while(p!=NULL){
            head = head->next;
            p=p->next;
        }
        return head;
    }
};
// @lc code=end

File: submitted_code/100294.lian-biao-zhong-dao-shu-di-kge-jie-dian-lcof.java
//
// @lc app=leetcode.cn id=100294 lang=java
//
// [100294] lian-biao-zhong-dao-shu-di-kge-jie-dian-lcof
//
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode getKthFromEnd(ListNode head, int k) {
        ListNode p = head;
        while(k>0){
            p = p.next;
            --k;
        }
        ListNode ans = head;
        while(p != null){
            ans = ans.next;
            p = p.next;
        }
        return ans;
    }
}
// @lc code=end